flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -503,6 +503,10 @@ private void initializeState(Collection<BufferedRequestState<RequestEntryT>> sta
         for (BufferedRequestState<RequestEntryT> state : states) {
             initializeState(state);
         }
+
+        if (bufferedRequestEntries.size() > 0 && !existsActiveTimerCallback) {
+            registerCallback();
+        }
     }
```

Comment:
Thanks for looking in to this @vahmed-hamdy . However I think we are duplicating logic here between `initializeState` and `addEntryToBuffer`. Instead of this, I propose we:

1. Delete the `initializeState` method
2. Overload `addEntryToBuffer` to include a version that accepts the `RequestEntryWrapper`, the existing one delegates to new wrapping the `RequestEntryT`
3. Update this method to something that just delegates to `addEntryToBuffer`, such as:
```
states.stream().flatMap(() -> state.getBufferedRequestEntries().stream()).forEach(entry -> addEntryToBuffer(...));
```

Reply:
Thanks @dannycranmer , I totally agree we have some duplicate logic between them methods.